simulation/traffic_simulator/include/traffic_simulator/hdmap_utils/cache.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2015-2020 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__HDMAP_UTILS__CACHE_HPP_
+#define TRAFFIC_SIMULATOR__HDMAP_UTILS__CACHE_HPP_
+
+#include <boost/optional.hpp>
+#include <geometry_msgs/msg/point.hpp>
+#include <mutex>
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/math/catmull_rom_spline.hpp>
+#include <unordered_map>
+#include <vector>
+
+namespace hdmap_utils
+{
+class RouteCache
+{
+public:
+  bool exists(std::int64_t from, std::int64_t to)
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    std::pair<std::int64_t, std::int64_t> key;
+    key.first = from;
+    key.second = to;
+    if (data_.find(key) == data_.end()) {
+      return false;
+    }
+    return true;
```

Comment:
This conditional expression is too verbose.
You can simply write `return data_.find (key)! = Data_.end ();`.

Reply:
I do not like ternary operator.

Reply:
THIS IS NOT TERNARY OPERATOR.

Reply:
OK, I understand, but it is not readable for me.